### DISCUSSION TITLE
`\time warp` according to counter numbers

### DISCUSSION THREAD

**Discussion by lucifiel1618** [gnembon/fabric-carpet#1558] [Ideas]
`\time warp` is very often used by people to measure efficiency of farms.
However, I usually see people put it as "The yield is approximately 10k items per hour, estimated by tick warping for 10 hours".
Users usually run tests unnecessarily long, because they can't have an idea of the yield from the very beginning of the test.

In addition, this is not the most scientific way to report results like this either, given the time for running test does not have direct statistical meaning. The better way would be "The yield approximately 10k items per hour within 1% uncertainty".
To this end, I think it would be super useful to add new ways to end `\time warp` with hopper counter inputs.

For example, `\time warp --counter white:"1%"` would end tick warping after you collect 10k items (equivalence of having 1% uncertainty in Poisson statistics) with the white wool counters. For the sake of completeness, there might also be `\time warp --counter white:10000`, which basically does the same thing.

Thank you.

**Comment by altrisi** [comment]
I think that would add a lot of complexity to the tick warp command that most people won't end up using, however you could do something like that with a scarpet app. You should be able to run the tick warp commands and every tick or few ticks check the counters to see if they match your requirements and then outputs your results.

**Reply by altrisi** [reply]
I'd say worst thing would be parsing the return of the counter command, but I think it should be doable. Feel free to ask for support on Discord if you want to go that route and need help.
